lms/static/js/ccx/view/ccx_schedule.js

```diff
@@ -0,0 +1,394 @@
+var define = window.define || RequireJS.define;  // jshint ignore:line
+
+define(
+  [
+    'backbone',
+    'jquery',
+    'underscore',
+    'gettext',
+    'moment',
+    'js/ccx/collection/schedule_collection',
+    'js/ccx/view/schedule_tree',
+    'js/ccx/view/schedule_right_container',
+    'text!templates/ccx/underscore/ccx_schedule.underscore'
+  ],
+  function (
+    Backbone,
+    $,
+    _,
+    gettext,
+    moment,
+    scheduleCollection,
+    scheduleTree,
+    scheduleRightContainerView,
+    scheduleTemplate
+  ) {
+    'use strict';
+    var ccxScheduleView = Backbone.View.extend({
+
+      initialize: function(options) {
+        this.saveCCXScheduleUrl = options.saveCCXScheduleUrl;
+        this.unSavedChanger = false;
+      },
+
+      render: function() {
+        this.scheduleJson = this.collection.toJSON();
+        this.$el.html(_.template(scheduleTemplate) ({}));
+        this.loadScheduleTree();
+        this.loadScheduleRightContainer();
+        return this;
+      },
+
+      loadScheduleRightContainer: function() {
+        // Right container consist of a form and alert messages.
+
+        // Hidden data will be shown into form,
+        // from there user can add this data to schedule tree
+        var scheduleFormData = this.pruned(this.scheduleJson, function(node) {
+          return node.hidden;
+        });
+
+        // load schedule right container view
+        var ccxScheduleFormDataCollection = new scheduleCollection.
+          ccxScheduleCollection(scheduleFormData);
+
+        var scheduleRightContainer = new scheduleRightContainerView({
+          el: this.$el.find("#ccx-schedule-form"),
+          collection: ccxScheduleFormDataCollection,
+          unSavedChanger: this.unSavedChanger,
+          parentView: this
```

Comment:
Instead of passing the parent view around, I've seen firing events from the children and listening to them on the parent to be more common and keeps things more decoupled.

That said, if you move those methods I said below around, you can just pass the collection around instead.